src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -544,4 +581,159 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+                        }else{
+                            result += nameString;
+                        }
+                    }
+                    lnCounter--;
+                    if (lnCounter > 0) {
+                        result += 'and ';
+                    }
+            }
+        }else{
+            for (String nameString : listNameStrings) {
+                if (contactFieldsNames.contains(nameString.toLowerCase())){
+                    Integer counter = cons.size();
+                    for (Contact con : cons) {
+                        result += con.get(nameString);
+                        counter--;
+                        if (counter > 0) {
+                            result += 'and ';
+                        }
+                    }
+                }else{
+                    result += nameString;
+                }
+            }
+        }
+        result = result.trim();
+        String output = settingPrefix + result + settingSuffix;
+        return output.trim();
+
+    }
+
+    /*******************************************************************************************************
+    * @description build the name Map of last name and contacts with the same last names
+    * @param cons list of sibling contacts
+    * @return map<String, List<Contact>. the map of last name and list of contacts
+    */
+    private static Map<String, List<Contact>> buildMapLastNameContacts(List<Contact> cons) {
+        Map<String, List<Contact>> mapLastNameContacts = new Map<String, List<Contact>>();
+        for (Contact con : cons)
+        {
+            if (mapLastNameContacts.containsKey(con.LastName)){
+                mapLastNameContacts.get(con.LastName).add(con);
+            }else{
+                mapLastNameContacts.put(con.LastName, new List<Contact>{con});
+            }
+        }
+        return mapLastNameContacts;
+    }
+
+    /*******************************************************************************************************
+    * @description update account names if any changes made to contacts' names or a new contact is linked the account
+    * @param accIdsToRename set of account ids whose name need to be updated
+    * @return void
+    */
+    private void updateHHAccountsName(Set<Id> accIdsToRename) {
+        list<Account> listAccsToRename = new list<Account>();
+
+        list<Account> listAcc = [SELECT Id,
+                                        RecordTypeId,
+                                        (SELECT FirstName,
+                                                LastName
+                                        FROM Contacts)
+                                FROM Account
+                                WHERE Id IN :accIdsToRename];
+
+        for (Account acc : listAcc) {
+            acc.Name = strNameHHAccountForContact(acc.Contacts);
+            listAccsToRename.add(acc);
+        }
+        // add all accounts needing updating to our list
+            update listAccsToRename;
```

Comment:
Instead of calling the update directly, can we merge the Accounts records in the dmlwrapper? 

---
        if (listAccountIdHHToUpdate.size() > 0) {
            updateParentAcc(listAccountIdHHToUpdate, dmlWrapper);
        }
---

Reply:
I thought we made the decision to leave this for now and work on dmlwrapper enhancements later because this work will take some additional resources given that we have limited bandwidth right now.

Reply:
oh, right. Can you make a comment like TO DO here. So we do not forgot why we do not use dmlwrapper?

Reply:
Absolutely, I will create a workitem if we haven't already created one.